src/esp/assets/ResourceManager.cpp

```diff
@@ -1149,23 +1151,42 @@ void ResourceManager::loadMaterials(Importer& importer,
     // as long as the materialName includes the full path to the material
     Cr::Containers::Optional<Mn::Trade::MaterialData> materialData =
         importer.material(iMaterial);
-    if (!materialData ||
-        !(materialData->types() & Magnum::Trade::MaterialType::Phong)) {
+
+    if (!materialData) {
       LOG(ERROR) << "Cannot load material, skipping";
       continue;
     }
 
-    const auto& phongMaterialData =
-        static_cast<Mn::Trade::PhongMaterialData&>(*materialData);
     std::unique_ptr<gfx::MaterialData> finalMaterial;
     int textureBaseIndex = loadedAssetData.meshMetaData.textureIndex.first;
-    if (loadedAssetData.assetInfo.requiresLighting) {
-      finalMaterial =
-          buildPhongShadedMaterialData(phongMaterialData, textureBaseIndex);
 
+    if (loadedAssetData.assetInfo.requiresLighting &&
+        materialData->types() &
+            Magnum::Trade::MaterialType::PbrMetallicRoughness) {
+      const Mn::Trade::PbrMetallicRoughnessMaterialData&
+          pbrMetallicRoughnessMaterialData =
+              static_cast<Mn::Trade::PbrMetallicRoughnessMaterialData&>(
+                  *materialData);
+
+      finalMaterial = gfx::buildPhongFromPbrMetallicRoughness(
+          pbrMetallicRoughnessMaterialData, textureBaseIndex, textures_);
     } else {
-      finalMaterial =
-          buildFlatShadedMaterialData(phongMaterialData, textureBaseIndex);
+      ASSERT(materialData);
+      if (!(materialData->types() & Magnum::Trade::MaterialType::Phong)) {
+        LOG(ERROR) << "Cannot load material, skipping";
+        continue;
+      }
+
+      const auto& phongMaterialData =
+          static_cast<Mn::Trade::PhongMaterialData&>(*materialData);
```

Comment:
This can be just the following now:

```suggestion
      const auto& phongMaterialData = materialData->as<Mn::Trade::PhongMaterialData>();
```